sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -278,11 +278,6 @@ case class AttributeReference(
     case _ => false
   }
 
-  override def semanticEquals(other: Expression): Boolean = other match {
-    case ar: AttributeReference => sameRef(ar)
-    case _ => false
-  }
-
```

Comment:
Don't need `canonicalized`?

Reply:
The default implementation works for `AttributeReference`